src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2738,6 +2738,16 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 }
             }
 
+            if (operand.Kind == BoundKind.DefaultPendingInference)
+            {
+                // PROTOTYPE(default) Need to handle type parameters here (`where T : struct` for instance)
+                if (!targetType.IsStructType() || targetType.IsNullableType())
+                {
```

Comment:
I didn't understand this comment. 
The handling of null above checks for `IsValueType`. So here I need to check for `IsStructType`.
`IsNullableType()` only checks if the type is `System.Nullable`